src/components/game/game.sagas.ts

```diff
@@ -38,31 +38,58 @@ function* moveMap(key: string) {
     }
 
     const tiles: TileMatrix = yield select(getGrid)
-    const { row: currentRow, col: currentCol } = yield select(getViewGrid)
+    const { row: currentRow, col: currentCol }: ViewGrid = yield select(getViewGrid)
 
     switch (key) {
         case 'ArrowUp':
-            return yield setViewGrid(tiles, currentRow - 1, currentCol)
+            return yield updateViewGrid(tiles, currentRow - 1, currentCol)
         case 'ArrowDown':
-            return yield setViewGrid(tiles, currentRow + 1, currentCol)
+            return yield updateViewGrid(tiles, currentRow + 1, currentCol)
         case 'ArrowRight':
-            return yield setViewGrid(tiles, currentRow, currentCol + 1)
+            return yield updateViewGrid(tiles, currentRow, currentCol + 1)
         case 'ArrowLeft':
-            return yield setViewGrid(tiles, currentRow, currentCol - 1)
+            return yield updateViewGrid(tiles, currentRow, currentCol - 1)
     }
 }
 
-function* setViewGrid(tiles: TileMatrix, row: number, col: number) {
+function* updateViewGrid(tiles: TileMatrix, row: number, col: number) {
     const viewSize: number = yield select(getViewSize)
 
-    if (row < 0 || col < 0 || row + viewSize === tiles.length || col + viewSize === tiles[0].length) {
+    if (row < 0 || col < 0 || row + viewSize > tiles.length || col + viewSize > tiles[0].length) {
         return
     }
+
     const viewTiles: TileMatrix = yield getViewTiles(tiles, viewSize, row, col)
     yield put(actions.setViewGrid({ row, col, grid: viewTiles }))
 }
 
+function* displayPlayerUnits(action: ReturnType<typeof actions.initPlayer>) {
+    const units = action.payload.units
+    if (!units) return
+
+    const currentTiles: TileMatrix = yield select(getGrid)
+    const tiles = currentTiles.map(row => row.slice())
+
+    units.forEach(unit => {
+        const tile = tiles[unit.position.row][unit.position.row]
+        tile.units = [...tile.units, unit]
+    })
+
+    yield put(actions.setGrid(tiles))
```

Comment:
setting a new grid every time a Unit is created should be fine. React will still use the existing DOM elements because they haven't changed (the `key` attribute helps React with this)